test/relations.test.js

```diff
@@ -918,6 +918,75 @@ describe('relations', function() {
       };
     });
 
+    describe('find over related model with options', function() {
+      after(function() {
+        Physician.clearObservers('access');
+        Patient.clearObservers('access');
+      });
+      before(function() {
+        Physician.observe('access', beforeAccessFn);
+        Patient.observe('access', beforeAccessFn);
+
+        function beforeAccessFn(ctx, next) {
+          ctx.query.where.realm = ctx.options.realm;
+          next();
+        }
+      });
+      it('should find be filtered from option', function(done) {
+        var id;
+        Physician.create(function(err, physician) {
+          physician.patients.create({name: 'a', realm: 'test'}, function(err, ch) {
+            id = ch.id;
+            physician.patients.create({name: 'z', realm: 'test'}, function() {
+              physician.patients.create({name: 'c', realm: 'anotherRealm'}, function() {
+                verify(physician);
+              });
+            });
+          });
+        });
+
+        function verify(physician) {
+          physician.patients({}, {realm: 'test'}, function(err, records) {
+            if (err) return done(err);
+            should.exist(records);
+            records.length.should.eql(2);
+            records[0].realm.should.eql('test');
```

Comment:
I think the assertion failure message produced by these checks will be rather unhelpful. I am proposing to use a different approach:

```js
should.exist(records);
const expected = ['a:test', 'z:test'];
const actual = records.map(function(r) { return r.name + ':' + r.realm; });
actual.should.eql(expected);
```